### PR TITLE
TAS-2354/translation-part-one

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

File: src/core/translation/i18n.ts [hunk 1]
```diff
@@ -3,19 +3,23 @@ import detector from 'i18next-browser-languagedetector';
...
+  },
 };
 
 i18next
   .use(detector)
   .use(initReactI18next)
   .init({
     resources,
-    lng: 'en',
+    lng: 'jp',
```
Comment (satanmourner) [inline src/core/translation/i18n.ts:22]: is init language japanese?

File: src/modules/Auth/containers/onboarding/Causes/useCauses.tsx [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 export const useCauses = () => {
-  const keytems = Object.keys(SOCIAL_CAUSES);
+  const keyItems = Object.keys(SOCIAL_CAUSES);
   const [value, setValue] = useState<SocialCauseVal[]>([]);
   const { state, updateUser } = useUser();
   const { updateSelectedStep } = useContext(StepsContext);
-  const items = keytems.map(i => {
-    return { value: SOCIAL_CAUSES[i].value, label: SOCIAL_CAUSES[i].label };
+  const items = keyItems.map(i => {
+    return { value: SOCIAL_CAUSES[i].value, label: translate(SOCIAL_CAUSES[i].value) };
```
Comment (satanmourner) [inline src/modules/Auth/containers/onboarding/Causes/useCauses.tsx:17]: i didn't check it, should it be `.value` for label?

File: src/modules/Auth/containers/onboarding/OrganizationCauses/index.tsx [hunk 3]
```diff
@@ -7,20 +8,22 @@ import { useOrganizationCauses } from './useOrganizationCauses';
...
         <MultiSelect
           id={'social-causes'}
-          searchTitle={'Select at least 1 cause*'}
+          searchTitle={translate('onboarding-org-cause-select-one')}
           max={5}
-          maxLabel={'Max. 5 causes'}
-          items={items}
-          placeholder={'Search a cause'}
+          maxLabel={translate('onboarding-org-cause-max')}
+          items={items.slice(0, 30)}
```
Comment (satanmourner) [inline src/modules/Auth/containers/onboarding/OrganizationCauses/index.tsx:25]: why `.slice(0,30)` ?

File: src/modules/Auth/containers/onboarding/Stepper/index.tsx [hunk 4]
```diff
@@ -30,15 +31,15 @@ const Steper: React.FC<Props> = ({ components }) => {
   const handleBack = () => {
     if (hasPrevStep) updateSelectedStep(step - 1);
   };
-
+  const { t } = useTranslation();
```
Comment (satanmourner) [inline src/modules/Auth/containers/onboarding/Stepper/index.tsx:34]: `translate`

File: src/pages/sign-up/Email/index.tsx [hunk 5]
```diff
@@ -22,21 +20,7 @@ export const Email = () => {
   const referrerUser = savedReferrer ? JSON.parse(savedReferrer) : null;
   const { tried } = useCaptcha();
   const navigate = useNavigate();
-  const events = (useLoaderData() as EventsRes) || null;
-  const [searchParams] = useSearchParams();
-  const eventName = searchParams.get('event_name') || '';
-  const eventId = events?.items.find(event => event.title === EVENTS_QUERIES[eventName])?.id || '';
-  const defaultIntro = {
```
Comment (satanmourner) [inline src/pages/sign-up/Email/index.tsx:29]: no event?

File: src/pages/sign-up/Congrats/index.tsx [hunk 6]
```diff
@@ -8,23 +9,24 @@ import { Stepper } from 'src/modules/general/components/stepper/stepper';
 
 export const Congrats = () => {
   const navigate = useNavigate();
+  const { t: translate } = useTranslation();
 
   const routeToNextPage = async () => {
     const path = await nonPermanentStorage.get('savedLocation');
-    navigate(path || '/sign-up/user/onboarding');
+    navigate(path ? path : '/sign-up/user/onboarding');
```
Comment (satanmourner) [inline src/pages/sign-up/Congrats/index.tsx:16]: path || 'string' => if path is falsy and has not value => 'string'

File: src/modules/general/components/profileCard/index.tsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ const ProfileCard: React.FC<ProfileCardProps> = ({ identity, labelShown = true,
   const myProfile = currentIdentity?.id === identity?.id;
   const navigate = useNavigate();
   const { name, profileImage, type, website, username } = getIdentityMeta(identity);
-  const socialCauses = socialCausesToCategory(identity?.social_causes).map(item => item.label);
+  const socialCauses = socialCausesToCategory(identity?.social_causes).map(item => translate(item.value));
   if (!identity) return;
   const events = eventsToCategory((identity as User).events || []).map(item => item.label);
```
Comment (satanmourner) [inline src/modules/general/components/profileCard/index.tsx:31]: i didn't get. do we have event or not?

**Review by satanmourner — CHANGES_REQUESTED** [review]

**Comment by satanmourner** [comment]
@marjan-ara please check failed test.

File: src/modules/Auth/containers/onboarding/Causes/useCauses.tsx (hunk 2 above)
Comment (marjan-ara) [reply src/modules/Auth/containers/onboarding/Causes/useCauses.tsx:17]: Yes, keys are used for translation

**Review by satanmourner — APPROVED** [review]

**Pull request merged**